### ISSUE TITLE
escape function does not return Markup class for instances implementing __html__ method

### ISSUE THREAD

**Issue by mila** [pallets/markupsafe#68]
The `escape` function does not wrap a return value of `__html__` method as `Markup`.

Expected behaviour documented in `README.rst`:
```
>>> class Foo(object):
...  def __html__(self):
...   return '<strong>Nice</strong>'
...
>>> escape(Foo())
Markup(u'<strong>Nice</strong>')
```

Actual behaviour:
```
>>> from markupsafe import escape
>>> class Foo(object):
...  def __html__(self):
...   return '<strong>Nice</strong>'
...
>>> escape(Foo())
'<strong>Nice</strong>'
```

I am using the 1.0 version I get same result for both native and speedups implementations. 
```
>>> import markupsafe
>>> markupsafe.__version__
'1.0'
>>> from markupsafe._speedups import escape
>>> escape(Foo())
'<strong>Nice</strong>'
>>> from markupsafe._native import escape
>>> escape(Foo())
'<strong>Nice</strong>'
```

**Comment by karinohka66** [comment]
Xauy

**Comment by dawran6** [comment]
Hi @mila , @davidism , I'm new to this project and would like your help.

I see the documentation for [New-Style String Formatting](https://github.com/pallets/markupsafe#new-style-string-formatting) explicitly return a `Markup` object instead of `str` object in the `__html__` method. Could it be the documentation that is wrong and instead it should say the following?
```python
>>> escape(Foo())
'<strong>Nice</strong>'
>>> Markup(Foo())
Markup(u'<strong>Nice</strong>')
```

**Comment by dawran6** [comment]
Ok I believe `escape` should always return the `Markup` object, for both `str` and custom class.

**Comment by mila** [comment]
@dawran6 This is a very good question. 

I decided to return `Markup` instance from my `__html__` method to be safe. 

On the other side I perceive the `__html__` method as an interface. The `Markup` class is one of possible implementations. It does not seem correct to make an interface depend on one specific implementation.

**Comment by dawran6** [comment]
Thanks @mila for the response. I agree with you.

**Comment by davidism** [comment]
fixed in #69
